### PR TITLE
Add the ability to redeem a token

### PR DESCRIPTION
**Pull request by conortwo**
Added form for redeeming tokens and related db logic. Crn it suggested you so we'll go with that :D

### CODE REVIEW

File: public/js/components.js [hunk 1]
```diff
@@ -159,4 +159,64 @@ class RequestForm extends React.Component {
...
+    }
+
+     // Checks if a token is valid
+    validateToken(token) {
+        let ref = firebase.database().ref('tokens/' + token);
+        ref.on('value', (s) => {
+            if(s.val() != null){
+                // If the token is in the db remove it and return it's value
+                if(s.val().redeemed) {
+                    this.state.errorMessage === 'Token was already redeemed!'
```
Comment (crnbrdrck) [inline public/js/components.js]: You're comparing the errorMessage here not assigning it :P

File: public/js/components.js [hunk 2]
```diff
@@ -159,4 +159,64 @@ class RequestForm extends React.Component {
...
+                    let reqs = s.val().value;
+                    ref.off();
+                    ref.update({
+                        redeemed: true
+                    });
+                    this.grantRequests(reqs);
+                }
+            }
+            else {
+                this.state.errorMessage === 'Unrecognised token, ensure it is entered correctly!';
```
Comment (crnbrdrck) [inline public/js/components.js]: Same here :P

File: public/js/components.js (hunk 1 above)
Comment (conortwo) [reply public/js/components.js]: derp, will fix! 😅 
I feel like we should be displaying these error messages, in a div next to the textbox or something?

Comment (crnbrdrck) [reply public/js/components.js]: I'll be adding that after I get brians changes :)

**Pull request merged**